### PR TITLE
feat: move the publication status logic in backend

### PR DESCRIPTION
**Pull request by ReidyT**

### CODE REVIEW

File: cypress/support/server.ts [hunk 1]
```diff
@@ -595,11 +596,7 @@ export const mockPostItemMembership = (
   cy.intercept(
     {
       method: HttpMethod.Post,
-      url: new RegExp(
-        `${API_HOST}/${parseStringToRegExp(
-          buildPostItemMembershipRoute(ID_FORMAT),
-        )}`,
-      ),
+      url: `${API_HOST}/item-memberships?*`,
```
Comment (spaenleh) [inline cypress/support/server.ts:599]: I think you need to escape the `?` no ? 
```suggestion
      url: `${API_HOST}/item-memberships\?*`,
```

File: package.json [hunk 2]
```diff
@@ -21,8 +21,8 @@
     "@emotion/styled": "11.11.5",
     "@graasp/chatbox": "3.1.0",
     "@graasp/map": "1.16.0",
-    "@graasp/query-client": "3.15.2",
-    "@graasp/sdk": "4.17.0",
+    "@graasp/query-client": "github:graasp/graasp-query-client#1115-determine-publication-state-backend",
+    "@graasp/sdk": "github:graasp/graasp-sdk#1115-determine-publication-state-backend",
```
Comment (spaenleh) [inline package.json:25]: Note to update

**Review by spaenleh — APPROVED** [review]
🚜 Looks good !

File: cypress/support/server.ts (hunk 1 above)
Comment (ReidyT) [reply cypress/support/server.ts:599]: No I don't think it is necessary:

`Unnecessary escape character: \?.eslint`

Comment (spaenleh) [reply cypress/support/server.ts:599]: Okay good then !

**Pull request merged**